feature_engine/discretisation/arbitrary.py

```diff
@@ -107,36 +120,32 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
         return self
 
     def transform(self, X: pd.DataFrame) -> pd.DataFrame:
-        """Sort the variable values into the intervals.
 
-        Parameters
-        ----------
-        X: pandas dataframe of shape = [n_samples, n_features]
-            The dataframe to be transformed.
+        X = super().transform(X)
 
-        Returns
-        -------
-        X_new: pandas dataframe of shape = [n_samples, n_features]
-            The transformed data with the discrete variables.
-        """
+        # check if NaN values were introduced by the discretisation procedure.
+        if X[self.variables_].isnull().sum().sum() > 0:
 
-        # check input dataframe and if class was fitted
-        X = super().transform(X)
+            # obtain the name(s) of the columns with null values
+            tmp = X[self.variables_].copy()
```

Comment:
@Morgan-Sell 

Can we remove creating a copy and just pass x[self.variables_] to the nan_columns below?

If the df is big, creating a copy may take up a lot of memory in the users computer.

Reply:
@solegalli 

If we delete `X = super().transform(X)`, three tests fail. See PyTest summary below:

1.  FAILED test_arbitrary_discretiser.py::test_arbitrary_discretiser - AssertionError: Attributes of DataFrame.iloc[:, 1] (column name="HouseAge") are different
2. FAILED test_arbitrary_discretiser.py::test_error_if_input_df_contains_na_in_transform - Failed: DID NOT RAISE <class 'ValueError'>
3. FAILED test_arbitrary_discretiser.py::test_error_when_nan_introduced_during_transform - Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) was emitted. The list of emitted warnings is: [].

Don't we need to use `X = super().transform()` to inherit the BaseDiscretiser's `transform()` properties?

Reply:
What I meant here is to delete this line: `tmp = X[self.variables_].copy()` and then to modify line 130 so that it is:

`nan_columns = X[self.variables_].columns[X[self.variables_].isnull().any()].tolist()`